src/goto-programs/goto_program_template.h

```diff
@@ -386,21 +386,12 @@ class goto_program_templatet
     return t;
   }
 
-  static const irep_idt get_function_id(
-    const_targett l)
-  {
-    while(!l->is_end_function())
-      ++l;
-
-    return l->function;
-  }
-
   static const irep_idt get_function_id(
     const goto_program_templatet<codeT, guardT> &p)
   {
-    assert(!p.empty());
+    PRECONDITION(!p.empty());
 
-    return get_function_id(--p.instructions.end());
+    return p.instructions.back().function;
```

Comment:
@tautschnig, I grab it from behind now. The only use of this is in dependency_graph.h for indexing the post_dominator map. There was another use in remove returns that was unnecessary (see below).

Reply:
That was probably my doing/my fault. Can we just get rid of it completely instead? I'm not even sure that `initialize` method of `dependence_grapht` is ever called. If it is, let's add another parameter that provides the function name.

Reply:
initialize overrides ait::initialize...

Reply:
We have
````c++
  virtual void initialize(const goto_programt &);
  virtual void initialize(const goto_functionst::goto_functiont &);
````
-> #1848.